libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/utils/gradleConfigurationUtils.kt

```diff
@@ -7,10 +7,10 @@ package org.jetbrains.kotlin.gradle.utils
 
 import org.gradle.api.Project
 
-fun Project.addExtendsFromRelation(extendingConfigurationName: String, extendsFromConfigurationName: String, forced: Boolean = true) {
```

Comment:
If `configurations.findByName(extendingConfigurationName) == null`, by default the existing code would still run `project.dependencies.add(extendingConfigurationName,...)` (which may succeed or fail depending on the contract of the `add` method, but in the new code it will just ignore it. Is the new behavior expected?

Reply:
`add` invokes `ConfigurationContainer.getByName(...)` which fails if that configuration does not exist i.e previous code:
- would not run if `force = false` if configuration does not exist; which is done in this PR
- would throw if `force = true` and conf does not exist => this was never the case.

Because of that, this change should not change the semantic of this code.

Reply:
> would throw if force = true and conf does not exist => this was never the case.

I agree this may not have happened, but the intention of `force` is to prevent this case from happening *in the future* right? So previously, calling `addExtendsFromRelation("nonExistentConfig", "existingConfig", force=true)` would fail, whereas the new code will just ignore it.

Reply:
I find it hard to justify usage of `force=true` TBH. Tests should ensure correctness of the implementation.

@h0tk3y Can you please chime in here, considering you are the original author of this utility?

Reply:
I think throwing is usually better than swallowing so normally we would write `configurations.findByName(extendingConfigurationName)!!.extendsFrom(...)` instead of `configurations.findByName(extendingConfigurationName)?.extendsFrom(...)`. But for some old projects as mentioned in the code comment we actually have to ignore the error, so maybe that's why `force` was introduced (and defaulted to `true`). Anyway, I'll defer to @h0tk3y on this.

Reply:
I've changed this to have a more helpful error message in case `forced=true`, failing early does make sense.